objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/imageRetriever/NetworkStreamRetriever.java

```diff
@@ -186,18 +188,27 @@ public ImageRetrieverResults getImage(Map<String, ?> request) throws ImageAcquis
         return results;
     }
     
-    public void diagnoseConnection() throws ImageAcquisitionException{
+    public void diagnoseConnection() throws ImageAcquisitionException {
         try {
-            URL urlProtocolTest = new URL((String) camera);
-            InputStream urlReadTest = urlProtocolTest.openStream();
-        } catch (MalformedURLException e) {
-//            throw new IllegalArgumentException(this.getClass().getCanonicalName() + ".unknownProtocol: "
-//                    + "URL specifies unknown protocol");
-            log.error(".unknownProtocol: Error parsing URL: ", e);
-            
-        } catch (java.io.IOException e) {
-            throw new ImageAcquisitionException(this.getClass().getCanonicalName() + ".badRead: "
-                    + "URL was unable to be read");
+            URI URITest = new URI(camera);
+            if (URITest.getScheme().equals("http") || URITest.getScheme().equals("https")) {
+                try {
+                  URL urlProtocolTest = new URL((String) camera);
+                  InputStream urlReadTest = urlProtocolTest.openStream();
+              } catch (MalformedURLException e) {
+                  throw new IllegalArgumentException(this.getClass().getCanonicalName() + ".unknownProtocol: "
```

Comment:
When basing an error or exception on another, better to include it.  So add 'e' as a another parameter on the end of these exceptions.  Otherwise, we know that we said XXX but we don't really know why.